### ISSUE TITLE
Packer build fails after disconnecting

### ISSUE THREAD

**Issue by jamestwebber** [hashicorp/packer#5829]
Creating a new issue after discussion in #5646 

In a few different cases, I have trouble finishing a packer build because (I believe) the instance is not responding often enough to keep the collection alive. This seems to happen when it's in the middle of e.g. compiling some large library.

From reading other issues it seems like this might be a problem in the Go ssh library, but I don't really know.

- `packer version` is 1.1.0
- Host platform is OS X High Sierra
- Debug log output from `PACKER_LOG=1 packer build template.json` is [here](https://gist.github.com/jamestwebber/a3628f3dfb181562defcd64ad589adda). It's quite long, but I think the salient part is:
```
[lots of compiling]
2018/01/30 19:11:48 ui:     amazon-ebs: gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o dict.o dict.c
    amazon-ebs: gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o dict.o dict.c
2018/01/30 19:11:48 ui:     amazon-ebs: gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o stats.o stats.c
    amazon-ebs: gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o stats.o stats.c
2018/01/30 19:12:34 packer: 2018/01/30 19:12:34 Remote command exited without exit status or exit signal.
2018/01/30 19:12:34 packer: 2018/01/30 19:12:34 [INFO] RPC endpoint: Communicator ended with: 2300218
```
Note that the disconnect is 45 seconds after the last message. My guess is that's the timeout limit.

- Unfortunately I am not sure what the simplest example to reproduce this is–by definition this happens only in a complex image. The image in the above log is [nanopore.json](https://github.com/czbiohub/packer-images/blob/specops/nanopore.json) and the relevant script is [nanopore.sh](https://github.com/czbiohub/packer-images/blob/specops/scripts/nanopore.sh)

Thanks in advance

**Comment by mwhooker** [comment]
Thanks for the report.

I believe this is solved in #5830, but I would appreciate your help testing if you're able to build that branch.

Alternatively, you could make sure your base AMI has the sshd_config settings:

```
ClientAliveInterval 300
ClientAliveCountMax 10
```

or similar, which should prevent it from disconnecting you.

**Comment by jamestwebber** [comment]
This appears to have fixed the problem, thank you! I tested the branch and `packer build` went off without a hitch in two different problematic cases.
